### PR TITLE
improve sync logic

### PR DESCRIPTION
**Pull request by vadiminc**

### CODE REVIEW

File: src/services/worker/tasks/burn-rates.task.ts [hunk 1]
```diff
@@ -68,6 +74,21 @@ export class BurnRatesTask {
...
+    );
+
+    const latestBlockNumber =
+      await this._web3Provider.web3.eth.getBlockNumber();
+
+    if (
+      latestSyncedBlockNumber + BurnRatesTask.BLOCK_RANGE <
+      latestBlockNumber
+    ) {
+      // this._logger.log(`Ignore task. Events are not fully synced yet.`);
```
Comment (axelrod-blox) [inline src/services/worker/tasks/burn-rates.task.ts:88]: if it unnecessary, let's remove it

File: src/services/worker/tasks/fetch.task.ts [hunk 2]
```diff
@@ -45,45 +42,62 @@ export class FetchTask {
...
+    );
+
+    const latestBlockNumber =
+      await this._web3Provider.web3.eth.getBlockNumber();
+
+    if (
+      latestSyncedBlockNumber + FetchTask.MAX_DELAY_BLOCK_RANGE >
+      latestBlockNumber
+    ) {
+      // this._logger.log(`Ignore task. Allowed delay range.`);
```
Comment (axelrod-blox) [inline src/services/worker/tasks/fetch.task.ts:56]: also here

File: src/services/worker/tasks/fetch.task.ts [hunk 3]
```diff
@@ -45,45 +42,62 @@ export class FetchTask {
...
+
+      const messageDetails = `From block: ${fromBlock}, to block: ${toBlock}, latest block: ${latestBlockNumber}`;
+      this._logger.log(
+        `Fetched all events till latest block. ${messageDetails}`,
+      );
+      /*
+      if (toBlock < latestBlockNumber) {
         this._logger.log(
-          `Sync completed for a blocks range: ${fromBlock} - ${toBlock}`,
+          `Continuing fetching immediately because latest block number is not reached. ${messageDetails}`,
```
Comment (axelrod-blox) [inline src/services/worker/tasks/fetch.task.ts:91]: we need it?

File: src/services/worker/tasks/liquidation.task.ts [hunk 4]
```diff
@@ -4,109 +4,146 @@ import { Cluster } from '@cli/modules/clusters/cluster.entity';
...
+    );
+
+    const latestBlockNumber =
+      await this._web3Provider.web3.eth.getBlockNumber();
+
+    if (
+      latestSyncedBlockNumber + LiquidationTask.BLOCK_RANGE <
+      latestBlockNumber
+    ) {
+      // this._logger.log(`Ignore task. Events are not fully synced yet.`);
```
Comment (axelrod-blox) [inline src/services/worker/tasks/liquidation.task.ts:44]: remove please

**Review by axelrod-blox — APPROVED** [review]

File: src/services/worker/cron/fetch.cron.ts [hunk 5]
```diff
@@ -7,7 +7,7 @@ import { FetchTask } from '../tasks/fetch.task';
 export class FetchCron {
   constructor(private _fetchTask: FetchTask) {}
 
-  @Cron('0 * * * * *')
+  @Cron('* * * * * *')
```
Comment (meshin-dev) [inline src/services/worker/cron/fetch.cron.ts:10]: Is it blocking operation, no few same crons will be executed at a time?

File: src/services/worker/tasks/burn-rates.task.ts (hunk 1 above)
Comment (meshin-dev) [reply src/services/worker/tasks/burn-rates.task.ts:88]: It should be uncommented as debug log

File: src/services/worker/tasks/fetch.task.ts (hunk 2 above)
Comment (meshin-dev) [reply src/services/worker/tasks/fetch.task.ts:56]: debug

File: src/services/worker/tasks/fetch.task.ts (hunk 3 above)
Comment (meshin-dev) [reply src/services/worker/tasks/fetch.task.ts:91]: @vadiminc if logger supports debug messages and by default level is higher, let's keep it as debugs and uncomment

**Review by meshin-dev — APPROVED** [review]
Nothing critical, I approve it, but @vadiminc @andrew-blox we should do sanity like @Or-blox did it, to go over different liquidation scenarios, before we merge it to stage and further

**Pull request merged**